src/beamlime/services/dashboard.py

```diff
@@ -201,17 +194,49 @@ def create_monitor_plot(key: str, data: sc.DataArray) -> go.Figure:
         )
         return fig
 
+    @staticmethod
+    def create_detector_plot(key: str, data: sc.DataArray) -> go.Figure:
+        if len(data.dims) == 1:
+            return DashboardApp.create_monitor_plot(key, data)
+
+        fig = go.Figure()
+        x_dim, y_dim = data.dims
+        fig.add_heatmap(
+            z=[[]],
+            x=[],  # Will be filled with coordinate values
+            y=[],  # Will be filled with coordinate values
+            colorscale='Viridis',
+        )
+
+        def maybe_unit(dim: str) -> str:
+            unit = data.coords[dim].unit
+            return f' [{unit}]' if unit is not None else ''
+
+        fig.update_layout(
+            title=key,
+            width=500,
+            height=400,
+            xaxis_title=f'{x_dim}{maybe_unit(x_dim)}',
+            yaxis_title=f'{y_dim}{maybe_unit(y_dim)}',
+            uirevision=key,
+        )
+        return fig
+
     def update_plots(self, n: int | None):
         if n is None:
             raise PreventUpdate
 
         try:
-            monitor_messages = self._source.get_messages()
-            num = len(monitor_messages)
-            monitor_messages = compact_messages(monitor_messages)
+            messages = self._source.get_messages()
+            num = len(messages)
+            messages = compact_messages(messages)
             self._logger.info(
-                "Got %d messages, showing most recent %d", num, len(monitor_messages)
+                "Got %d messages, showing most recent %d", num, len(messages)
             )
+            monitor_messages = [
+                msg for msg in messages if 'beam_monitor' in msg.key.topic
+            ]
+            detector_messages = [msg for msg in messages if 'detector' in msg.key.topic]
```

Comment:
Can the looping over messages become expensive at any point?
I.e. is it worth refactoring to have a single `for msg in messages` and splitting between monitor and detector inside the loop?

Reply:
There are typically O(10) messages.